Mathlib/MeasureTheory/Constructions/Convolution.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.MeasureTheory.Constructions.Prod.Basic
+import Mathlib.MeasureTheory.Measure.MeasureSpace
+/-!
+# The convolution of measures
+
+In this file we define and prove properties about the convolution of two measures
+
+## Main definition
+
+* `MeasureTheory.Measure.conv`: The convolution of two measures.
+
+## Main results
+
+* add
+## Implementation Notes
+
+## Tags
+
+-/
+
+noncomputable section
+
+namespace MeasureTheory
+
+namespace Measure
+
+/-- Convolutions of measures. They are defined for arbitrary measures on a monoid M that is also
+a measurable space. TODO: should get a to_additive version for AddMonoids -/
+def conv {M : Type*} [Monoid M] [MeasurableSpace M] (μ : Measure M) (ν : Measure M) :
+    Measure M := Measure.map (fun x : M × M ↦ x.1 * x.2) (Measure.prod μ ν)
+
+/-- Scoped notation for the convolution of measures. -/
+scoped[MeasureTheory] infix:80 " * " => MeasureTheory.Measure.conv
+
+-- Convolution of the dirac measure at 1 with a measure μ returns μ. -/
+theorem one_convolution {M : Type*} [Monoid M] [MeasurableSpace M] [MeasurableMul₂ M]
```

Comment:
Since the definition is called `conv`, you should use `conv` instead of `convolution` in all the lemma names.

What about `dirac_one_conv` for this one?

Reply:
I'll change that!